bin/uploadZip.js

```diff
@@ -16,14 +16,30 @@ const request = require('request-promise-native');
 const getReactorHeaders = require('./getReactorHeaders');
 const handleResponseError = require('./handleResponseError');
 
-module.exports = async (envConfig, accessToken, extensionPackageId, zipPath) => {
-  const isNew = !extensionPackageId;
+module.exports = async (envConfig, accessToken, extensionPackageManifest, extensionPackageFromServer, zipPath) => {
+  const shouldPost = !extensionPackageFromServer || extensionPackageFromServer.attributes.availability !== 'development';
+
+  if (extensionPackageFromServer) {
+    if (extensionPackageFromServer.attributes.availability === 'development') {
+      console.log(`An existing extension package with the name ` +
+        `${chalk.bold(extensionPackageManifest.name)} was found on the server and will be updated. ` +
+        `The extension package ID is ${chalk.bold(extensionPackageFromServer.id)}.`);
+    } else {
+      console.log(`An existing extension package with the name ` +
+        `${chalk.bold(extensionPackageManifest.name)} was found on the server, but because its ` +
+        `availability is not ${chalk.bold('development')}, a development version of the extension package ` +
+        `will be created.`);
+    }
+  } else {
+    console.log(`No extension package was found on the server with the ` +
```

Comment:
It seems to me that this branch is not tested.

Reply:
The "uploads a zip for a new extension package" test covers this.

Reply:
Nevermind, it doesn't assert that the message is actually logged. I'll add that.